src/VisualStudio/Core/Def/Implementation/Watson/WatsonReporter.cs

```diff
@@ -43,10 +42,36 @@ public static void InitializeFatalErrorHandlers(TelemetrySession session)
             compilerNonFatalErrorHandlerProperty.SetValue(null, nonFatalReporter);
         }
 
-        public static void InitializeLogger(TraceSource logger)
+        public static void RegisterTelemetrySesssion(TelemetrySession session)
         {
-            Debug.Assert(s_logger == null);
-            s_logger = logger;
+            lock (_guard)
+            {
+                s_telemetrySessions = s_telemetrySessions.Add(session);
+            }
+        }
+
+        public static void UnregisterTelemetrySesssion(TelemetrySession session)
+        {
+            lock (_guard)
+            {
+                s_telemetrySessions = s_telemetrySessions.Remove(session);
+            }
+        }
+
+        public static void RegisterLogger(TraceSource logger)
+        {
+            lock (_guard)
+            {
+                s_loggers.Add(logger);
```

Comment:
s_loggers is an ImmutableArray.  so this doesn't do anything (afaict).

Reply:
good catch

Reply:
still seems to apply.